lib/ex_doc/formatter/html.ex

```diff
@@ -128,7 +128,14 @@ defmodule ExDoc.Formatter.HTML do
 
   def ast_to_html({tag, attrs, ast}) do
     attrs = Enum.map(attrs, fn {key, val} -> " #{key}=\"#{val}\"" end)
-    ["<#{tag}", attrs, ">", ast_to_html(ast), "</#{tag}>"]
+
+    case tag do
+      tag when tag in [:pre, :code] ->
+        ["<#{tag}", attrs, ">", Enum.intersperse(ast_to_html(ast), "\n"), "</#{tag}>"]
```

Comment:
I see. My question though is: why are we breaking .txt over multiple lines? Shouldn't we keep it as a single binary? I am afraid we are adding new interpretations to the AST while not necessary. We can also add the newlines after splitting in the splitting code instead of here.

Reply:
> I see. My question though is: why are we breaking .txt over multiple lines? Shouldn't we keep it as a single binary? I am afraid we are adding new interpretations to the AST while not necessary. We can also add the newlines after splitting in the splitting code instead of here.

yes, that is because of Earmark's and the new ExDoc's AST that I have been working on. That is how it is represented, because there can be tags inside a `<pre>`. I will simplify this for now.

Reply:
fixed!